crowbar_framework/app/models/service_object.rb

```diff
@@ -1421,6 +1432,25 @@ def proposal_dependencies(role)
     []
   end
 
+  def expand_items_in_elements(elements)
+    # expand items in elements that are not nodes
+    expanded_new_elements = {}
+    failures = []
+    elements.each do |role_name, nodes|
+      expanded_new_elements[role_name], failures = expand_nodes_for_all(nodes)
+      unless failures.nil? || failures.empty?
+        Rails.logger.fatal(
+          "apply_role: Failed to expand items #{failures.inspect} for role \"#{role_name}\""
+        )
+        message = "Failed to apply the proposal: cannot expand list of nodes for role \"#{role_name}\", following items do not exist: #{failures.join(", ")}"
+        Rails.logger.progress("Failed to apply role #{role.name}")
+        update_proposal_status(inst, "failed", message)
+        return [expanded_new_elements, failures, message]
```

Comment:
To make it even more visible it's an error, I would return `[nil, failures, message]`

Reply:
done! message removed altogether